google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/spi/DefaultBigQueryRpc.java

```diff
@@ -80,7 +80,7 @@
   private final Bigquery bigquery;
 
   public DefaultBigQueryRpc(BigQueryOptions options) {
-    HttpTransportOptions transportOptions = options.getHttpTransportOptions();
+    HttpTransportOptions transportOptions = (HttpTransportOptions) options.getTransportOptions();
```

Comment:
This wouldn't be null - we would check for which type it was. 

Yes, the naming of `Default*Rpc` is a problem - I will take care of renaming it in a following PR; I also want to remove the parameterization of `ServiceRpcT`, because essentially `BigQueryOptions` is currently typed with `DefaultBigQueryRpc` (`HttpBigQueryRpc`), meaning http hasn't been removed from the hierarchy yet.